daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc.hs

```diff
@@ -625,17 +626,31 @@ execMigrate opts inFile1 inFile2 mbDir = do
     forM_ pairs $ \(e1, e2) -> do
         let path1 = eRelativePath e1
         let path2 = eRelativePath e2
-        let generatedPath =
+        let upgradeModPath =
                 joinPath $ fromMaybe "" mbDir : (tail $ splitPath path1)
+        let instancesModPath1 =
+                replaceBaseName upgradeModPath $
+                takeBaseName path1 <> "InstancesA"
+        let instancesModPath2 =
+                replaceBaseName upgradeModPath $
+                takeBaseName path2 <> "InstancesB"
         opts' <- Compiler.mkOptions opts
         parsedMod1 <- parse opts' loggerH path1
         parsedMod2 <- parse opts' loggerH path2
-        let generatedMod =
+        let generatedUpgradeMod =
                 generateUpgradeModule
                     (pkg1, pm_parsed_source parsedMod1)
                     (pkg2, pm_parsed_source parsedMod2)
-        createDirectoryIfMissing True $ takeDirectory generatedPath
-        writeFile generatedPath generatedMod
+        let generatedInstancesMod1 =
+                generateGenInstancesModule "A" (pkg1, pm_parsed_source parsedMod1)
+        let generatedInstancesMod2 =
+                generateGenInstancesModule "B" (pkg2, pm_parsed_source parsedMod2)
+        forM [ (upgradeModPath, generatedUpgradeMod)
```

Comment:
Should be `forM_`?

Reply:
right, wonder why I didn't got a warning. fixed.